utils/cards.spec.ts

```diff
@@ -0,0 +1,36 @@
+import "child_process";
+import { setupCardsStorage, updateCardsStorage } from "./cards";
+
+let execHistory: string[] = [];
+
+jest.mock("child_process", () => ({
+  exec: (command: string, cb: (_: null, result: string) => void) => {
+    execHistory.push(command);
+    cb(null, command);
+  }
+}));
+
+describe("cardsUtils", () => {
+  beforeEach(() => {
+    execHistory = [];
+  });
+  describe("setupCardsStorage", () => {
+    it("remove old and clone new dir", async () => {
```

Comment:
It's a nitpick, but should be `removes old and clone new dir`.

Reply:
Also I've checked and this testsuit doesn't works on my computer (i think it depends on the platform).
I believe it's will be fine if we add tests about platform.
I've roughly written smth like this (also is needed to declare a `originalPlatform` variable):
```
 beforeEach(() => {
    originalPlatform = Object.getOwnPropertyDescriptor(process, 'platform');
    execHistory = [];
  });

  afterEach(() => {
    Object.defineProperty(process, 'platform', originalPlatform);
  });

  describe("setupCardsStorage", () => {
    it("removes old and clone new dir if platform is linux", async () => {
      Object.defineProperty(process, 'platform', {
        value: 'linux'
      });
      await setupCardsStorage();
      expect(execHistory).toEqual(
        expect.arrayContaining([
          "rm -rf data/cards",
          "git clone https://github.com/memory-cards/cards data/cards --depth 2"
        ])
      );
    });
``` 
and add the same for `win32`
Also as I understand if we test function and `process` variable we should get process value exactly at this function. So working case for me if at file `cards.ts` we will move `isWin` and `rmCommand` inside the function.
```
export const setupCardsStorage = () => {
  const isWin = process.platform.includes("win") && process.platform !== "darwin";

  // in *nix `rm` by default skip non-existing files,
  // on win* we have to use another command and check if directory exists
  const rmCommand = isWin
    ? `if exist "data\\cards" rmdir /s /q "data\\cards"`
    : "rm -rf data/cards";

  return Promise.resolve()
    .then(() => exec(rmCommand))
    .then(() =>
      exec(
        "git clone https://github.com/memory-cards/cards data/cards --depth 2"
      )
    );
}
```
What do you think about it?

Reply:
It removes directory only on start - we can't run just `git pull` cause we're not sure about existing directory

we could use git-submodules, if you're about it
---
not sure if it worth to add platform specific logic to tests, better wrap to docker container, I suppose

Reply:
I will think about gitsubmodules